x-pack/plugins/alerting/common/rule_task_instance.ts

```diff
@@ -22,8 +22,16 @@ export const ruleStateSchema = t.partial({
   previousStartedAt: t.union([t.null, DateFromString]),
 });
 
+const ruleExecutionMetricsSchema = t.partial({
+  numQueries: t.number,
+  totalSearchDurationMs: t.number,
+  totalQueryDurationMs: t.number,
+});
+
+export type RuleExecutionMetrics = t.TypeOf<typeof ruleExecutionMetricsSchema>;
 export type RuleTaskState = t.TypeOf<typeof ruleStateSchema>;
-export type RuleTaskStateWithActions = RuleTaskState & {
+export type RuleExecutionState = RuleTaskState & {
```

Comment:
Renamed from `RuleTaskStateWithActions` because I didn't want to make it `RuleTaskStateWithActionsAndMetrics` :). Used `RuleExecutionState` because it is the state that is returned after rule execution, which is then transformed into `RuleTaskState` before being persisted in the task document.